File: android/app/src/main/kotlin/com/app/driver/LocationEventStreamHandler.kt
package com.app.driver

import android.os.Handler
import android.os.Looper
import io.flutter.plugin.common.EventChannel

class LocationEventStreamHandler : EventChannel.StreamHandler {

    companion object {
        private var eventSink: EventChannel.EventSink? = null
        private val mainHandler = Handler(Looper.getMainLooper())

        fun sendLocation(locationData: HashMap<String, Any>) {
            mainHandler.post {
                eventSink?.success(locationData)
            }
        }
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }
}

File: ios/Runner/LocationEventStreamHandler.swift
import Flutter

class LocationEventStreamHandler: NSObject, FlutterStreamHandler {

    static let shared = LocationEventStreamHandler()

    private var eventSink: FlutterEventSink?

    func sendLocation(_ locationData: [String: Any]) {
        DispatchQueue.main.async { [weak self] in
            self?.eventSink?(locationData)
        }
    }

    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        eventSink = events
        return nil
    }

    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        eventSink = nil
        return nil
    }
}
